tests/e2e/test_response_caching.py

```diff
@@ -227,7 +228,8 @@ def handler() -> None: ...
         cur = cur.app
     unpacked_middleware.append(cur)
 
-    assert len([m for m in unpacked_middleware if isinstance(m, ResponseCacheMiddleware)]) == int(expect_applied)
+    # if no cache the outermost middleware is ExceptionMiddleware and we'll get MethodType instead
```

Comment:
Why? Not immediately obvious to me

Reply:
hopefully this one is more explicit, I test now if cache in `@get` is true that we have a ResponseCacheMiddleware and if not just the ExceptionHandlerMiddleware